src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -0,0 +1,426 @@
+          size,
+          SnowflakeSinkConnectorConfig.BUFFER_SIZE_BYTES_DEFAULT);
+      this.fileSize = SnowflakeSinkConnectorConfig.BUFFER_SIZE_BYTES_DEFAULT;
+    } else {
+      this.fileSize = size;
+      LOGGER.info("set buffer size limitation to {} bytes", size);
+    }
+  }
+
+  @Override
+  public void setTopic2TableMap(Map<String, String> topic2TableMap) {
+    this.topic2TableMap = topic2TableMap;
+  }
+
+  @Override
+  public void setFlushTime(long time) {
+    if (time < SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC_MIN) {
+      LOGGER.error(
+          "flush time is {} seconds, it is smaller than the minimum "
+              + "flush time {} seconds, reset to the minimum flush time",
+          time,
+          SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC_MIN);
+      this.flushTime = SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC_MIN;
+    } else {
+      this.flushTime = time;
+      LOGGER.info("set flush time to {} seconds", time);
+    }
+  }
+
+  @Override
+  public void setMetadataConfig(SnowflakeMetadataConfig configMap) {
+    this.recordService.setMetadataConfig(configMap);
+  }
+
+  @Override
+  public long getRecordNumber() {
+    return this.recordNum;
+  }
+
+  @Override
+  public long getFlushTime() {
+    return this.flushTime;
+  }
+
+  @Override
+  public long getFileSize() {
+    return this.fileSize;
+  }
+
+  @Override
+  public void setBehaviorOnNullValuesConfig(
+      SnowflakeSinkConnectorConfig.BehaviorOnNullValues behavior) {
+    this.behaviorOnNullValues = behavior;
+  }
+
+  @Override
+  public void setCustomJMXMetrics(boolean enableJMX) {
+    this.enableCustomJMXMonitoring = enableJMX;
+  }
+
+  @Override
+  public SnowflakeSinkConnectorConfig.BehaviorOnNullValues getBehaviorOnNullValuesConfig() {
+    return this.behaviorOnNullValues;
+  }
+
+  @Override
+  public void setDeliveryGuarantee(
+      SnowflakeSinkConnectorConfig.IngestionDeliveryGuarantee ingestionDeliveryGuarantee) {
+    this.ingestionDeliveryGuarantee = ingestionDeliveryGuarantee;
+  }
+
+  @Override
+  public Optional<MetricRegistry> getMetricRegistry(String pipeName) {
+    return Optional.empty();
+  }
+
+  /**
+   * Gets a unique identifier consisting of topic name and partition number.
+   *
+   * @param topic topic name
+   * @param partition partition number
+   * @return combinartion of topic and partition
+   */
+  @VisibleForTesting
+  protected static String partitionChannelKey(String topic, int partition) {
+    return topic + "_" + partition;
+  }
+
+  // ------ Streaming Ingest Related Functions ------ //
+  private OpenChannelRequest getOpenChannelRequest(
+      final String channelName, final String tableName) {
+    return OpenChannelRequest.builder(channelName)
+        .setDBName(this.connectorConfig.get("snowflake.database.name"))
+        .setSchemaName(this.connectorConfig.get("snowflake.schema.name"))
+        .setTableName(tableName)
```

Comment:
is the table name is fully qualified table name or without db and schema?

Reply:
And we also add a required onErrorOption field, you probably want to update the jar file you use

Reply:
> is the table name is fully qualified table name or without db and schema?

No, only table name since DB and schema are added explicitly in config. And, table name is not provided, `tablename = f(topicName)` and there are some rules which we will stick with. 

https://docs.snowflake.com/en/user-guide/kafka-connector-install.html#required-privileges

Reply:
> And we also add a required onErrorOption field, you probably want to update the jar file you use

Will do. Thanks.